### PR TITLE
rough draft of the foundation guide, using Cata BDK as a guinea pig

### PR DESCRIPTION
**Pull request by emallson**
this adds a basic *Foundation Guide* that uses core analyzers to produce some basic analysis with minimal additional code.

the goal is for you to be able to take this and plug it in to a previously-unsupported spec to get something useful just by setting up the core analyzers.

i ended up punting on the resource section for now because we don't have any truly core (in the sense that they operate off of the Config or Abilities) pieces for that just yet. However, I think that having a shorthand to set up `ResourceTracker` / `ResourceGraph` will be a good solution.

(the `App.scss` changes are mostly whitespace. i dropped the line height a little because we're doing more prose)

### TODOs

- [x] Melee DPS / Tank
- [x] Caster DPS (https://github.com/WoWAnalyzer/WoWAnalyzer/pull/6838)
- [x] Healer (https://github.com/WoWAnalyzer/WoWAnalyzer/pull/6841)
  - [x] Healer Mana section
- [x] Wrap in top-level "Core Skills" section
- [ ] Maybe adjust the HL color / remove HLs? i don't love it

### Preview

image

### Misc Other Changes

- Added a warning about food/flask logging in Cata Classic
- Small tweak to the `Highlight` component

### CODE REVIEW

File: src/interface/guide/foundation/FoundationCooldownSection.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+  if (!cooldowns || cooldowns.length === 0) {
+    return null;
+  }
+
+  return (
+    <Section title="Cooldown Usage">
+      <Explanation>
+        <p>
+          Perfect cooldown usage is a kind of art&mdash;a combination of in-depth fight knowledge
```
Comment (jazminite) [inline src/interface/guide/foundation/FoundationCooldownSection.tsx]:
```suggestion
          Perfect cooldown usage is a combination of in-depth fight knowledge
```

File: src/interface/guide/foundation/FoundationCooldownSection.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+          and player skill to get exceptional output. However, 90% of the time you can get 90% of
```
Comment (jazminite) [inline src/interface/guide/foundation/FoundationCooldownSection.tsx]:
```suggestion
          and player skill. However, 90% of the time you can get 90% of
```

File: src/interface/guide/foundation/FoundationCooldownSection.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+          those results by <strong>making sure to use every cooldown available to you.</strong>
+        </p>
+        <p>
+          The key idea is to <HL>use your cooldowns as many times as you can.</HL> For example: you
```
Comment (jazminite) [inline src/interface/guide/foundation/FoundationCooldownSection.tsx]:
```suggestion
          The key idea is to <HL>use your cooldowns as many times as you can.</HL> For example, you
```

File: src/interface/guide/foundation/FoundationDowntimeSection.tsx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+  const info = useInfo();
+  if (!alwaysBeCasting) {
+    return null;
+  }
+
+  return (
+    <Section title="Always Be Casting">
+      <Explanation>
+        <p>
+          The foundation of good play in <em>World of Warcraft</em> is having low downtime: step 1
```
Comment (jazminite) [inline src/interface/guide/foundation/FoundationDowntimeSection.tsx]:
```suggestion
          The foundation of good play in <em>World of Warcraft</em> is having low downtime.
```

File: src/interface/guide/foundation/FoundationDowntimeSection.tsx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+                There should be no gaps between the end of one spell cast and the start of the next.
+              </Role.Caster>
+            </ByRole>
+          </HL>{' '}
+          It is better to use the wrong spell and keep going than it is to stop and think between
+          each cast&mdash;using nothing does no damage or healing, but using anything (even if it
+          isn't the <em>best</em> choice) will at least do <em>something.</em>
+        </p>
+        <p>
+          With practice, you will be able to both keep active and pick the right spells for each
```
Comment (jazminite) [inline src/interface/guide/foundation/FoundationDowntimeSection.tsx]:
```suggestion
          With practice, you can keep active and pick the right spells for each
```

File: src/interface/guide/foundation/FoundationDowntimeSection.tsx [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+      </p>
+      <p>
+        <ActiveTimeGraph
+          activeTimeSegments={alwaysBeCasting.activeTimeSegments}
+          fightStart={info!.fightStart}
+          fightEnd={info!.fightEnd}
+        />
+      </p>
+      <p>
+        A good rule of thumb is that{' '}
```
Comment (jazminite) [inline src/interface/guide/foundation/FoundationDowntimeSection.tsx]:
```suggestion
        As a general guideline,{' '}
```

File: src/interface/guide/foundation/FoundationDowntimeSection.tsx [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+  if (!alwaysBeCasting) {
+    return null;
+  }
+
+  return (
+    <Section title="Always Be Casting">
+      <Explanation>
+        <p>
+          The foundation of good play in <em>World of Warcraft</em> is having low downtime: step 1
+          to doing well is to <strong>Always Be Casting</strong>.{' '}
```
Comment (jazminite) [inline src/interface/guide/foundation/FoundationDowntimeSection.tsx]:
```suggestion
          The first step is to <strong>Always Be Casting (ABC)</strong>.{' '}
```

**Review by jazminite — CHANGES_REQUESTED** [review]
Wording suggestions :)

**Comment by emallson** [comment]
those suggestions all look good. gonna make them locally so that prettier doesn't complain

**Comment by emallson** [comment]
before merging this, i want to also implement it for a caster dps and a healer. gonna update the OP with a checklist

**Review by jazminite — APPROVED** [review]
Wording updates look good!

**Pull request merged**